core/src/View/Kokkos_ViewMapping.hpp

```diff
@@ -1763,35 +1771,38 @@ struct ViewOffset<
-                                                                                       7
-                                                                                   ? size_t(
-                                                                                         1)
-                                                                                   : m_dim
-                                                                                         .N7)))))))) {
+            arg_layout.stride != KOKKOS_IMPL_CTOR_DEFAULT_ARG
+                ? arg_layout.stride
+                : Padding<TrivialScalarSize>::
+                      stride(/* 2 <= rank */
+                             m_dim.N1 *
+                             (dimension_type::rank == 2
+                                  ? size_t(1)
+                                  : m_dim.N2 *
+                                        (dimension_type::rank == 3
+                                             ? size_t(1)
+                                             : m_dim.N3 *
+                                                   (dimension_type::rank == 4
+                                                        ? size_t(1)
+                                                        : m_dim.N4 *
+                                                              (dimension_type::
+                                                                           rank ==
+                                                                       5
+                                                                   ? size_t(1)
+                                                                   : m_dim.N5 *
+                                                                         (dimension_type::
+                                                                                      rank ==
+                                                                                  6
+                                                                              ? size_t(
+                                                                                    1)
+                                                                              : m_dim.N6 *
+                                                                                    (dimension_type::
+                                                                                                 rank ==
+                                                                                             7
+                                                                                         ? size_t(
+                                                                                               1)
+                                                                                         : m_dim
+                                                                                               .N7)))))))) {
```

Comment:
Can you just introduce a free function that you call here to make this somewhat readable?

Reply:
done - not that I think it helps much